# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.cpp
# Commit: 2186bd51bebd
# Full Hash: 2186bd51bebd0a087d2b37b58ddb590110eda236
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 4 - Move ObjectFlags from BaseShape to Shape. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106974
# ==============================================================================

diff -r d5d3cb487914 -r 2186bd51bebd js/src/jit/MacroAssembler.cpp
--- a/js/src/jit/MacroAssembler.cpp	Sat Mar 06 01:05:20 2021 +0000
+++ b/js/src/jit/MacroAssembler.cpp	Sat Mar 06 01:05:20 2021 +0000
@@ -3621,12 +3621,11 @@
 void MacroAssembler::branchIfObjectNotExtensible(Register obj, Register scratch,
                                                  Label* label) {
   loadPtr(Address(obj, JSObject::offsetOfShape()), scratch);
-  loadPtr(Address(scratch, Shape::offsetOfBaseShape()), scratch);
 
   // Spectre-style checks are not needed here because we do not interpret data
   // based on this check.
   static_assert(sizeof(ObjectFlags) == sizeof(uint16_t));
-  load16ZeroExtend(Address(scratch, BaseShape::offsetOfFlags()), scratch);
+  load16ZeroExtend(Address(scratch, Shape::offsetOfObjectFlags()), scratch);
   branchTest32(Assembler::NonZero, scratch,
                Imm32(uint32_t(ObjectFlag::NotExtensible)), label);
 }